linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/rule.yml

```
@@ -37,19 +44,21 @@ references:
     nist: CM-6(a)
     srg: SRG-OS-000105-GPOS-00052,SRG-OS-000375-GPOS-00160,SRG-OS-000375-GPOS-00161,SRG-OS-000377-GPOS-00162
     stigid@rhel7: RHEL-07-041001
+    stigid@sle12: SLES-12-030500
     stigid@rhel8: RHEL-08-010390
 
 ocil_clause: 'smartcard software is not installed'
 
-{{%- if product in ["rhel7", "ol7"] %}}
-ocil: '{{{ ocil_package(package="pam_pkcs11") }}}'
-{{%- else %}}
-ocil: '{{{ ocil_package(package="openssl-pkcs11") }}}'
-{{%- endif %}}
+ocil: |-
+    {{% for pkg in smartcard_packages %}}
+    '{{{ ocil_package(package=pkg) }}}'
+    {{% endfor %}}
 
+{{% if product != 'sle12' %}}
 template:
     name: package_installed
     vars:
         pkgname: openssl-pkcs11
         pkgname@rhel7: pam_pkcs11
         pkgname@ol7: pam_pkcs11
+{{% endif %}}
```

when using the specialized files in the `ansible` and `oval` folder... for example the `ansible/sle12.yml` will automatically use the specialized one. In this case, I think you can remove this Jinja if and replace it with:

```suggestion
template:
    name: package_installed
    vars:
        pkgname: openssl-pkcs11
        pkgname@rhel7: pam_pkcs11
        pkgname@ol7: pam_pkcs11
    backends:
        bash@sle12: "off"
```

---

No need to do this change right now. We may in the future revert the template usage for this rule because it can happen that for RHEL8 STIG this might contain more than one package and the template cannot handle multiple packages. For now we can leave like this.